src/client/api/event.c

```diff
@@ -1031,6 +1053,12 @@ daos_event_fini(struct daos_event *ev)
 		if (eqx == NULL)
 			return -DER_NONEXIST;
 		eq = daos_eqx2eq(eqx);
+		D_MUTEX_LOCK(&eqx->eqx_lock);
+	}
+
+	if (evx->evx_status == DAOS_EVS_RUNNING) {
+		rc = -DER_BUSY;
+		goto out;
```

Comment:
is it reasonable to fini an event in RUNNING?

Reply:
well the problem there is that when we fini the event in running, DAOS does not know that, and when RPC completes, it calls daos_event_complete on that event, and then we get segfault

Reply:
but to answer you question, yes it can happen. In Jiafu's case, he is calling fini on events that may not have completed and that is causing assertion in his Java code.
We don't have cancellation mechanism yet from DAOS API, and after looking at it, it will be hard to support, since each event can have many underlying rpc requests that could be in parallel or chain of dependency. this just does the appropriate thing for now to return an error on an inflight event.